inmem_transport.go

```diff
@@ -178,9 +180,11 @@ func (i *InmemTransport) makeRPC(target ServerAddress, args interface{}, r io.Re
 		if rpcResp.Error != nil {
 			err = rpcResp.Error
 		}
-	case <-time.After(timeout):
+	case <-timer.C:
 		err = fmt.Errorf("command timed out")
 	}
+
+	timer.Stop()
```

Comment:
I think it's currently safe to call Stop here, but in the future we could add other returns. It might be good to keep this as a `defer` after creating the timer, both for consistency, and to prevent future regressions. Is there a reason that `defer` won't work in this case?

Reply:
I agree with you. 

Not using `defer` here is just a trade-off between safety (future changes) and performance (tiny improvement). Obviously, not calling `timer.Stop` would result in the same manner as leaking `time.After`.

I would fix that as your suggestion. Thank you so much.